content/mysql/concepts/built-in-functions/built-in-functions.md

```diff
@@ -0,0 +1,821 @@
+('{"name": "Santiago", "age": 25}');
+```
+
+As a result, the query will return the names stored in the JSON documents in the 'person_info' column:
+
+```shell
+#Query to extract the name from the JSON document
+SELECT JSON_EXTRACT(person_info, '$.name') AS Name FROM people;
+```
+
+| Name      |
+|-----------|
+| "Nelson"  |
+| "Santiago" |
+
+
+### Syntax
+
+`JSON_ARRAY()` creates a JSON array from a list of values.
+
+`JSON_ARRAY(value1, value2, ...)`
+
+### Example
+
+In the following example, there are five columns (one of them is an auto_increment) that correspond to one queue.
+The table **example_data** below will take these different data type values from the columns and compress them in a list of values as JSONArray.
+
+Below a complete example using MySQL code:
+
+```sql
+CREATE TABLE example_data (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    value1 INT,
+    value2 INT,
+    value3 VARCHAR(50),
+    value4 INT
+);
+
+INSERT INTO example_data (value1, value2, value3, value4) VALUES
+(1, 2, 'three', 4);
+```
+
+The example results in the following output:
+
+```shell
+#Query to create a JSON array from the inserted values
+SELECT JSON_ARRAY(value1, value2, value3, value4) AS JSONArray FROM example_data;
+```
+
+| JSONArray          |
+|--------------------|
+| [1, 2, "three", 4] |
+
+
+
+## Encryption and Compression Functions
+
+These functions provide encryption and compression utilities.
+
+### Syntax
+
+`AES_ENCRYPT()` encrypts a string using AES (Advanced Encryption Standard) encryption, and returns the original (binary) in [hexadecimal format](https://www.codecademy.com/resources/docs/general/hexadecimal).
+
+`AES_ENCRYPT(str, key_str)`
+
+### Example
+
+In the following example, there are 3 columns (one of them is an auto_increment) that correspond to one queue.
+The table **secrets** below will take the value *SecretText* from **plain_text** and will represent it in a binary code of [hexadecimal encrypted message](https://www.codecademy.com/resources/docs/general/hexadecimal).
+
+Below a complete example using MySQL code:
+
+```sql
+CREATE TABLE secrets (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    plain_text VARCHAR(50),
+    encrypted_text VARBINARY(50)
+);
+
+INSERT INTO secrets (plain_text, encrypted_text) VALUES
+('SecretText', AES_ENCRYPT('SecretText', 'key'));
+```
+
+The example results in the following output:
+
+```shell
+-- Query to select the encrypted text
+SELECT plain_text, AES_ENCRYPT(plain_text, 'key') AS EncryptedText FROM secrets;
+```
+
+| plain_text | EncryptedText                   |
+|------------|---------------------------------|
+| SecretText | (encrypted text representation) |
+
+> [!NOTE]
+> When dealing with encrypted data, we usually store the result in a 'VARBINARY' column because the encrypted data is binary. For display purposes, we might convert it to a readable format using functions like ['HEX()'](https://www.codecademy.com/resources/docs/python/built-in-functions/hex) if needed.
```

Comment:
```suggestion
> **Note:** When dealing with encrypted data, we usually store the result in a 'VARBINARY' column because the encrypted data is binary. For display purposes, we might convert it to a readable format.
```